modules/tensor_mechanics/examples/piston/piston_params.i

```diff
@@ -0,0 +1,152 @@
+## This example is documented on YouTube at:
+## https://www.youtube.com/watch?v=L9plLYkAbvQ
+## 
+## Additional files (e.g. the CAD model, results)
+## can be downloaded freely from Zenodo at:
+## https://doi.org/10.5281/zenodo.3886965
+
+[GlobalParams]
+  displacements = 'disp_x disp_y disp_z'
+[]
+
+[Mesh]
+  type = FileMesh #Read in mesh from file
+  file = piston_coarse.e
+[]
+
+# This is where mesh adaptivity magic happens:
+[./Adaptivity]
+  steps = 1
+  #initial_steps = 2
+  max_h_level = 3
+  cycles_per_step = 1
+  initial_marker = uniform
+  marker = errorFraction
+  [./Markers]
+    [./uniform]
+      type = UniformMarker
+      mark = refine
+    [../]
+    
+    [./errorFraction]
+      type = ErrorFractionMarker
+      coarsen = 0.5
+      indicator = gradientJump
+      refine = 0.5
+    [] 
+ [../]
+
+  [./Indicators]
+    [./gradientJump]
+      type = GradientJumpIndicator
+      variable = disp_y
+    []
+  []
+[../]
+
+[Variables]
+  [./disp_x]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./disp_y]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./disp_z]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[Modules/TensorMechanics/Master]
+      # parameters that apply to all subblocks are specified at this level. They
+      # can be overwritten in the subblocks.
+      add_variables = true
+      strain = FINITE
+      generate_output = 'vonmises_stress'
+      [./block]
+      block = 1
+      []
+  []
+
+[BCs]
+  [./Pressure]
+    [./load]
+      #Applies the pressure
+      boundary = load_surf
+      function = 't*550e5'
+      displacements = 'disp_x disp_y disp_z'
+    [../]
+  [../]
+  [./symmetry_x]
+    #Applies symmetry on the xmin faces
+    type = DirichletBC
+    variable = disp_x
+    boundary = 'xmin'
+    value = 0.0
+  [../]
+  [./hold_y]
+    #Anchors the bottom against deformation in the y-direction
+    type = DirichletBC
+    variable = disp_y
+    boundary = 'ymin'
+    value = 0.0
+  [../]
+  [./symmetry_z]
+    #Applies symmetry on the zmin faces
+    type = DirichletBC
+    variable = disp_z
+    boundary = 'zmin'
+    value = 0.0
+  [../]
+[]
+
+[Materials]
+  [./elasticity_tensor_steel]
+    #Creates the elasticity tensor using steel parameters
+    youngs_modulus = 210e9 #Pa
+    poissons_ratio = 0.3
+    type = ComputeIsotropicElasticityTensor
+    block = 1
+  [../]
+  [./stress]
+    #Computes the stress, using linear elasticity
+    type = ComputeFiniteStrainElasticStress
+    block = 1
+  [../]
+  [./density_steel]
+    #Defines the density of steel
+    type = GenericConstantMaterial
+    block = 1
+    prop_names = density
+    prop_values = 7850 # kg/m^3
+  [../]
```

Comment:
do we need `density`?

Reply:
We don't _need_ it for this simulation.  I'd left it in because I'd been testing some dynamics.  I can remove.

Reply:
See https://github.com/idaholab/moose/pull/18543/commits/e36103757a45619ae4457914d27d693054582e9a